src/include/ikarus/Geometries/GeometryType.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+#include <iostream>
+
+// Further types should be added in the future
+namespace Ikarus {
+  enum class GeometryType {
+    vertex,
+    linearLine,
+    quadraticLine,
+    cubicLine,
+    linearTriangle,
+    linearQuadrilateral,
+    linearTetrahedron,
+    pyramid,
+    prism,
+    linearHexahedron
+  };
+
+  inline bool isVertex(GeometryType type) { return type == Ikarus::GeometryType::vertex; }
+  inline bool isLinearLine(GeometryType type) { return type == Ikarus::GeometryType::linearLine; }
+  inline bool isQuadraticLine(GeometryType type) { return type == Ikarus::GeometryType::quadraticLine; }
+  inline bool isCubicLine(GeometryType type) { return type == Ikarus::GeometryType::cubicLine; }
+  inline bool isLinearTriangle(GeometryType type) { return type == Ikarus::GeometryType::linearTriangle; }
+  inline bool isLinearQuadrilateral(GeometryType type) { return type == Ikarus::GeometryType::linearQuadrilateral; }
+  inline bool isLinearTetrahedron(GeometryType type) { return type == Ikarus::GeometryType::linearTetrahedron; }
+  inline bool isPyramid(GeometryType type) { return type == Ikarus::GeometryType::pyramid; }
+  inline bool isPrism(GeometryType type) { return type == Ikarus::GeometryType::prism; }
+  inline bool isLinearHexahedron(GeometryType type) { return type == Ikarus::GeometryType::linearHexahedron; }
+
+  inline int dimension(GeometryType type) {
+    if (isVertex(type)) {
+      return 0;
+    } else if (isLinearLine(type) || isQuadraticLine(type) || isCubicLine(type)) {
+      return 1;
+    } else if (isLinearTriangle(type) || isLinearQuadrilateral(type)) {
+      return 2;
+    } else if (isLinearTetrahedron(type) || isPyramid(type) || isPrism(type) || isLinearHexahedron(type)) {
+      return 3;
+    } else {
+      throw std::runtime_error(
+          "Dimension of the given type couldn't be obtained. Check function 'dimension' in GeometryType.h");
+    }
+  }
+  // LCOV_EXCL_START
+  inline std::string print(GeometryType type) {
+    switch (type) {
+      case GeometryType::vertex:
+        return "vertex";
+      case GeometryType::linearLine:
+        return "linearLine";
+      case GeometryType::quadraticLine:
+        return "quadraticLine";
+      case GeometryType::cubicLine:
+        return "cubicLine";
+      case GeometryType::linearTriangle:
+        return "linearTriangle";

```

Comment:
Here, clang-format changed your original onliners to this case seperate from return statement.

There, is a clang-format switch "AllowShortCaseLabelsOnASingleLine: true" which can be added to https://github.com/IkarusRepo/Ikarus/blob/master/.clang-format

Then the to blocks look like:
```cpp
  inline int dimension(GeometryType type) {
    switch (type) {
      case GeometryType::vertex: return 0;
      case GeometryType::linearLine:
      case GeometryType::quadraticLine:
      case GeometryType::cubicLine: return 1;
      case GeometryType::linearTriangle:
      case GeometryType::linearQuadrilateral: return 2;
      case GeometryType::linearTetrahedron:
      case GeometryType::pyramid:
      case GeometryType::prism:
      case GeometryType::linearHexahedron: return 3;
      default:
        throw std::runtime_error(
            "Dimension of the given type couldn't be obtained. Check function 'dimension' in GeometryType.h");
    }
  }
  // LCOV_EXCL_START
  inline std::string print(GeometryType type) {
    switch (type) {
      case GeometryType::vertex: return "vertex";
      case GeometryType::linearLine: return "linearLine";
      case GeometryType::quadraticLine: return "quadraticLine";
      case GeometryType::cubicLine: return "cubicLine";
      case GeometryType::linearTriangle: return "linearTriangle";
      case GeometryType::linearQuadrilateral: return "linearQuadrilateral";
      case GeometryType::linearTetrahedron: return "linearTetrahedron";
      case GeometryType::pyramid: return "pyramid";
      case GeometryType::prism: return "prism";
      case GeometryType::linearHexahedron: return "linearHexahedron";
      default:
        throw std::runtime_error(
            "ostream operator for this GeometryType is not implemented. Check implementation in GeometryType.h");
    }
  }
```
I think the multiline version is more readable and we should not add the AllowShortCaseLabelsOnASingleLine flag

Reply:
If we keep it like it is the result is:

```cpp
  inline int dimension(GeometryType type) {
    switch (type) {
      case GeometryType::vertex:
        return 0;
      case GeometryType::linearLine:
      case GeometryType::quadraticLine:
      case GeometryType::cubicLine:
        return 1;
      case GeometryType::linearTriangle:
      case GeometryType::linearQuadrilateral:
        return 2;
      case GeometryType::linearTetrahedron:
      case GeometryType::pyramid:
      case GeometryType::prism:
      case GeometryType::linearHexahedron:
        return 3;
      default:
        throw std::runtime_error(
            "Dimension of the given type couldn't be obtained. Check function 'dimension' in GeometryType.h");
    }
  }
  // LCOV_EXCL_START
  inline std::string print(GeometryType type) {
    switch (type) {
      case GeometryType::vertex:
        return "vertex";
      case GeometryType::linearLine:
        return "linearLine";
      case GeometryType::quadraticLine:
        return "quadraticLine";
      case GeometryType::cubicLine:
        return "cubicLine";
      case GeometryType::linearTriangle:
        return "linearTriangle";
      case GeometryType::linearQuadrilateral:
        return "linearQuadrilateral";
      case GeometryType::linearTetrahedron:
        return "linearTetrahedron";
      case GeometryType::pyramid:
        return "pyramid";
      case GeometryType::prism:
        return "prism";
      case GeometryType::linearHexahedron:
        return "linearHexahedron";
      default:
        throw std::runtime_error(
            "ostream operator for this GeometryType is not implemented. Check implementation in GeometryType.h");
    }
  }
```

Reply:
The two line version is ok for me